### PR TITLE
Add macAddress configuration for template

### LINKED ISSUE #226: Add the ability to set the MAC address when starting a container
We have a docker image that requires a static MAC address due to a node lock license requirement. I have tried to use the LXC Conf Options to set the MAC address with no luck. 

I changed the daemon to use the lxc driver (--exec-driver=lxc) and set the LXC option to
lxc.network.hwaddr=02:d0:c6:0a:29:33

It seems to just be ignored, so we went back to using the libcontainer driver (default)

So, to try to fix this,  I upgraded my fork to use version 1.3 of the docker-java dependency which introduced a withMacAddress method to the dockerjava/api/command/CreateContainerCmd.java class.

When I modify the DockerTemplateBase class of the plugin and set containerConfig.withMacAddress("02:d0:c6:0a:29:33"), the container starts with that MAC Address. 

So, here are my questions:

    1) Is there a way to set the container before starting to a static MAC address with the current plugin
         version? 
    2) If the answer to 1 is no, are there any plans to upgrade the plugin to use version 1.3 of docker-java?
    3) If the answer to 2 is yes, would adding a MAC Address field to the plugin config panel make sense?

Basically, we really need a way to set a static MAC address when starting a container from the plugin configuration panel that works.

### PR DESCRIPTION
**Pull request by KostyaSha**
Closes #226

### CODE REVIEW

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java [hunk 1]
```diff
@@ -279,6 +295,10 @@ public CreateContainerCmd fillContainerConfig(CreateContainerCmd containerConfig
         if (environment != null && environment.length > 0)
             containerConfig.withEnv(environment);
 
+        if (getMacAddress() != null && !getMacAddress().trim().isEmpty()) {
```
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: `StringUtils.isNotBlank()`

Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: I don't like guava functions internals. Jenkins has Utils.* And all input variables handling need to be reviewed later. So this not critical atm.

Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: its apache commons-lang, not guava. Jenkins's Utils.* mostly copy-paste from there

Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: changed

**Comment by dcheese** [comment]
Looks good. The only suggestion would be to change the name to "MAC Address" within the config panel. Other than this,  +1 to merge

**Comment by KostyaSha** [comment]
@dcheese changed, thanks

**Comment by dcheese** [comment]
So, when will the next release be available that includes this (docker-plugin-0.9.0-rc2)?

**Comment by KostyaSha** [comment]
@dcheese i think it will be 0.9.0 because volumes was the main blocker. There are many different not blocking issues so i think it makes sense to do periodic releases RORE! 
So will do new release soon.

**Comment by dcheese** [comment]
great, thanks!

**Comment by dcheese** [comment]
soon, like a week or a month?

**Comment by KostyaSha** [comment]
@dcheese today, just doing release and fixing for other plugin.

**Comment by KostyaSha** [comment]
```
[INFO] Uploaded: http://maven.jenkins-ci.org:8081/content/repositories/releases/com/nirima/docker-plugin/0.9.0/docker-plugin-0.9.0.hpi (11573 KB at 1079.6 KB/sec)
```

**Comment by dcheese** [comment]
just one last thing, I noticed that there was a default MAC address set instead of it defaulting an empty string/blank. Is that what you intended? If it not set, then it should generate a MAC address which I think should be the default behavior. Thoughts?

**Pull request merged**